### PR TITLE
feat: LT-1622 - Restrict Roles and Permissions Visible to User

### PR DESCRIPTION
**Pull request by cstobart**

### CODE REVIEW

File: users/forms/users.py [hunk 1]
```diff
@@ -21,14 +22,15 @@ def add_user_form(request):
     )
 
 
-def edit_user_form(request, user_id, super_user):
+def edit_user_form(request, user_id, cannot_edit_role: Boolean):
```
Comment (Mark-Spiers-kainos) [inline users/forms/users.py]: was wondering if you could just use bool? and not import

File: users/forms/users.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 from django.urls import reverse_lazy
+from setoptconf import Boolean
```
Comment (stephanieboyd) [inline users/forms/users.py]: Do we already install this package?

File: users/forms/users.py (hunk 1 above)
Comment (cstobart) [reply users/forms/users.py]: good point

File: users/forms/users.py (hunk 2 above)
Comment (cstobart) [reply users/forms/users.py]: good point

File: users/forms/users.py [hunk 3]
```diff
@@ -21,14 +20,15 @@ def add_user_form(request):
     )
 
 
-def edit_user_form(request, user_id, super_user):
+def edit_user_form(request, user_id, cannot_edit_role: bool):
     return Form(
         title="Edit User",
         questions=[
             TextInput(title="Email", name="email"),
             Select(name="team", title="What team will the user belong to?", options=get_teams(request, True)),
```
Comment (stephanieboyd) [inline users/forms/users.py]: In the interest of tidying as we go - move to strings? Sorry for not spotting it the first time!

File: users/forms/users.py [hunk 4]
```diff
@@ -21,14 +20,15 @@ def add_user_form(request):
     )
 
 
-def edit_user_form(request, user_id, super_user):
+def edit_user_form(request, user_id, cannot_edit_role: bool):
```
Comment (andytaylorkainos) [inline users/forms/users.py]: can we invert the boolean please? (`can_edit_role`) and drop the `not` in the line 30 below

File: users/views/users.py [hunk 5]
```diff
@@ -65,17 +65,19 @@ class EditUser(TemplateView):
     def get(self, request, **kwargs):
         user, _ = get_gov_user(request, str(kwargs["pk"]))
         super_user = is_super_user(user) and request.user.lite_api_user_id == str(kwargs["pk"])
-        return form_page(request, edit_user_form(request, str(kwargs["pk"]), super_user), data=user["user"])
+        cannot_edit_role = user["user"]["id"] == request.user.lite_api_user_id or super_user
```
Comment (andytaylorkainos) [inline users/views/users.py]: please invert the boolean

Comment (cstobart) [reply users/views/users.py]: done

File: users/views/users.py [hunk 6]
```diff
@@ -65,7 +65,7 @@ class EditUser(TemplateView):
     def get(self, request, **kwargs):
         user, _ = get_gov_user(request, str(kwargs["pk"]))
         super_user = is_super_user(user) and request.user.lite_api_user_id == str(kwargs["pk"])
-        can_edit_role = (user["user"]["id"] != request.user.lite_api_user_id) or not super_user
+        can_edit_role = not (user["user"]["id"] == request.user.lite_api_user_id or super_user)
```
Comment (andytaylorkainos) [inline users/views/users.py]: `can_edit_role = user["user"]["id"] != request.user.lite_api_user_id`

**Pull request merged**